File: backend/internal/dto/announcement.go
package dto

import "time"

// --- Announcement DTOs ---

type CreatePollOptionReq struct {
	Text         string `json:"text" binding:"required"`
	DisplayOrder int    `json:"display_order"`
}

type CreatePollReq struct {
	Question      string                `json:"question" binding:"required"`
	IsMultiChoice bool                  `json:"is_multiple_choice"`
	IsAnonymous   bool                  `json:"is_anonymous"`
	Deadline      *time.Time            `json:"deadline"`
	Options       []CreatePollOptionReq `json:"options" binding:"required,min=2"`
}

type CreateAnnouncementReq struct {
	Title      string         `json:"title" binding:"required"`
	Content    string         `json:"content" binding:"required"`
	TargetType string         `json:"target_type" binding:"required,oneof=all team project"`
	TargetID   *uint          `json:"target_id"`
	IsPinned   bool           `json:"is_pinned"`
	ExpiresAt  *time.Time     `json:"expires_at"`
	Poll       *CreatePollReq `json:"poll"`
}

type UpdateAnnouncementReq struct {
	Title      string     `json:"title"`
	Content    string     `json:"content"`
	TargetType string     `json:"target_type" binding:"omitempty,oneof=all team project"`
	TargetID   *uint      `json:"target_id"`
	IsPinned   *bool      `json:"is_pinned"`
	ExpiresAt  *time.Time `json:"expires_at"`
}

type AnnouncementFilter struct {
	TargetType string `form:"target_type"`
	Page       int    `form:"page,default=1"`
	Size       int    `form:"size,default=20"`
}

// --- Poll Vote DTOs ---

type SubmitVoteReq struct {
	OptionIDs []uint `json:"option_ids" binding:"required,min=1"`
}

// --- Poll Results Response ---

type PollOptionResult struct {
	ID         uint    `json:"id"`
	Text       string  `json:"text"`
	VoteCount  int     `json:"vote_count"`
	Percentage float64 `json:"percentage"`
}

type VoterInfo struct {
	EmployeeID   uint   `json:"employee_id"`
	EmployeeName string `json:"employee_name"`
	OptionID     uint   `json:"option_id"`
	OptionText   string `json:"option_text"`
}

type PollResultsResponse struct {
	PollID     uint               `json:"poll_id"`
	Question   string             `json:"question"`
	TotalVotes int                `json:"total_votes"`
	IsAnonymous bool              `json:"is_anonymous"`
	IsClosed   bool               `json:"is_closed"`
	MyVotes    []uint             `json:"my_votes"`
	Options    []PollOptionResult `json:"options"`
	Voters     []VoterInfo        `json:"voters,omitempty"`
}

File: mobile/lib/features/announcements/domain/entities/announcement.dart
import 'package:equatable/equatable.dart';

class PollOption extends Equatable {
  const PollOption({
    required this.id,
    required this.pollId,
    required this.text,
    required this.voteCount,
    required this.displayOrder,
    this.percentage = 0.0,
  });

  final int id;
  final int pollId;
  final String text;
  final int voteCount;
  final int displayOrder;
  final double percentage;

  @override
  List<Object?> get props => [id, pollId, text, voteCount, displayOrder];
}

class Poll extends Equatable {
  const Poll({
    required this.id,
    required this.announcementId,
    required this.question,
    required this.isMultipleChoice,
    required this.isAnonymous,
    required this.status,
    this.deadline,
    this.options = const [],
    this.myVotes = const [],
    this.totalVotes = 0,
  });

  final int id;
  final int announcementId;
  final String question;
  final bool isMultipleChoice;
  final bool isAnonymous;
  final String status;
  final String? deadline;
  final List<PollOption> options;
  final List<int> myVotes;
  final int totalVotes;

  bool get isClosed => status == 'closed';
  bool get hasVoted => myVotes.isNotEmpty;

  bool get isDeadlinePassed {
    if (deadline == null) return false;
    final dl = DateTime.tryParse(deadline!);
    if (dl == null) return false;
    return dl.isBefore(DateTime.now());
  }

  @override
  List<Object?> get props => [id, announcementId, question, status, myVotes];
}

class Announcement extends Equatable {
  const Announcement({
    required this.id,
    required this.title,
    required this.content,
    required this.targetType,
    required this.createdAt,
    this.targetId,
    this.isPinned = false,
    this.expiresAt,
    this.poll,
  });

  final int id;
  final String title;
  final String content;
  final String targetType;
  final String createdAt;
  final int? targetId;
  final bool isPinned;
  final String? expiresAt;
  final Poll? poll;

  @override
  List<Object?> get props => [id, title, targetType, createdAt, isPinned];
}
